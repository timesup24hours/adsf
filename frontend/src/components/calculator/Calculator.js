import React from 'react'

const Calculator = props => {
  return (
    <div className='Calculator-container'>
      <div className='Calculator-button'>
        <button
          className='waves-effect waves-light btn blue'
          onClick={() => console.log('Calculator.js place order implementation needed!')} >
          Place your order
        </button>
      </div>
      <div className='Calculator-holder'>
        <div className='Calculator-div Calculator-order-summary'>Order Summary</div>
        <div className='Calculator-div'>
          <div>Items ({props.cart.totalQuantity}):</div>
          <div>$ {props.cart.total}</div>
        </div>
        <div className='Calculator-div Calculator-shipping-handling'>
          <div>Shipping & handling:</div>
          <div>$ </div> {/* shipping fee calcu function needed */}
        </div>
        <div className='Calculator-divider'>
          <div></div>
        </div>
        <div className='Calculator-div'>
          <div>Total before tax:</div>
          <div>$ </div> {/* total + shipping */}
        </div>
        <div className='Calculator-div'>
          <div>Estimated tax to be collected:</div>
          <div>$ </div> {/* tax */}
        </div>
        <div className='Calculator-divider'>
          <div></div>
        </div>
        <div className='Calculator-div Calculator-total-summary'>
          <div>Order Total:</div>
          <div>$</div> {/* order total */}
        </div>
      </div>
    </div>
  )
}

export default Calculator
