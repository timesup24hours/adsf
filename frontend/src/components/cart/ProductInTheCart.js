import React, { Component } from 'react'
import { connect } from 'react-redux'
// import Add from 'material-ui/svg-icons/content/add'
// import Remove from 'material-ui/svg-icons/content/remove'
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import * as cartActions from '../../store/actions/cartActions'
import { Link } from 'react-router'
// import { sliceContent } from '../../utils'

export class ProductInTheCart extends Component {
  state = {
    open: false,
  }

  handleOpen = () => {
    this.setState({ open: true });
  }

  handleClose = () => {
    this.setState({ open: false });
  }

  handleAddQuantity = () => {
    this.props.increaseQuantityOfProductInTheCart({ productId: this.props.product.product._id })
  }

  handleSubtractQuantity = () => {
    this.props.subtractQuantityOfProductIntheCart({ productId: this.props.product.product._id })
  }

  handleConfirmRemoveItemFromCart = () => {
    this.props.removeProductFromCart({ productId: this.props.product.product._id })
    this.handleClose()
  }

  render() {
    const actions = [
      <FlatButton
        label='No'
        primary={true}
        onTouchTap={this.handleClose}
      />,
      <FlatButton
        id='ProductInTheCart-remove'
        label='Yes'
        primary={true}
        keyboardFocused={true}
        onTouchTap={this.handleConfirmRemoveItemFromCart}
      />,
    ]
    const { images, _id, name, price, soldBy, deleted  } = this.props.product.product
    return (
      <div className='ProductInTheCart-container'>

        <div className='ProductInTheCart-info-container'>

          {deleted
            ? <div className='ProductInTheCart-product-deleted flexCenter'>
                <div>this product has been deleted by seller</div>
              </div>
            : null
          }
          <div className='ProductInTheCart-img'>
            {/* without s3
                <img src={`/images/products/${images[0]}`} alt=""/>
              */}
            <img src={images[0]} alt=""/>
          </div>
          <div className='ProductInTheCart-info-holder'>
            <div className='ProductInTheCart-div ProductInTheCart-info-name'>
              <Link to={`/product/${_id}`}>{name}</Link>
            </div>
            <div className='ProductInTheCart-div ProductInTheCart-info-price'>${price}</div>
            {/*<div className='ProductInTheCart-div ProductInTheCart-info-desc'>{sliceContent(desc, 100)}</div>*/}

            <div className='ProductInTheCart-div ProductInTheCart-info-quantity-section'>
              <div className='ProductInTheCart-info-quantity'><strong>Quantity:</strong> {this.props.product.quantity}</div>
              <span id='ProductInTheCart-add' className='ProductInTheCart-info-quantity-change glyphicon glyphicon-plus' aria-hidden="true" onClick={this.handleAddQuantity}></span>
              {this.props.product.quantity === 1
                ? null
                : <span id='ProductInTheCart-substract' className='ProductInTheCart-info-quantity-change glyphicon glyphicon-minus' onClick={this.handleSubtractQuantity}></span>
              }{/*? <div className='ProductInTheCart-info-quantity-change' onClick={this.handleOpen}>remove</div>*/}
            </div>
            <div className='ProductInTheCart-info-seller text-flow'>Sold by : {soldBy || ''}</div>
          </div>
        </div>

        <div className='ProductInTheCart-right-session'>
          <button
            id='ProductInTheCart-open-dialog'
            className='btn btn-danger'
            onClick={this.handleOpen}>
            Remove
          </button>
        </div>

        <Dialog
          id='ProductInTheCart-Dialog'
          title={this.props.totalQuantity === 1 ? 'Remove and Redirect Confirmation' : 'Remove Confirmation'}
          actions={actions}
          modal={false}
          open={this.state.open}
          onRequestClose={this.handleClose}
        >
          {this.props.totalQuantity === 1
            ? <div>Are you want to remove the last item from the cart, and Redirect to the Shop Page</div>
            : <div>Are you want to remove this item from the cart?</div>
          }
        </Dialog>

      </div>
    )
  }

}

ProductInTheCart.propTypes = {
  totalQuantity: React.PropTypes.number.isRequired,
}

const mapStateToProps = state => ({
  totalQuantity: state.cart.totalQuantity,
})

const mapDispatchToProps = dispatch => ({
  // changeQuantityOfProductIntheCart: payload => dispatch(cartActions.changeQuantityOfProductIntheCart(payload)),
  subtractQuantityOfProductIntheCart: payload => dispatch(cartActions.subtractQuantityOfProductIntheCart(payload)),
  increaseQuantityOfProductInTheCart: payload => dispatch(cartActions.increaseQuantityOfProductInTheCart(payload)),
  removeProductFromCart: payload => dispatch(cartActions.removeProductFromCart(payload)),
})

export default connect(mapStateToProps, mapDispatchToProps)(ProductInTheCart)
