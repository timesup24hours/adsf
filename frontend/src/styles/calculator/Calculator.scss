.Calculator-container {
  max-width: 300px;
  min-width: 300px;
  height: 320px;
  display: flex;
  flex-direction: column;
  padding: 14px 16px;
  font-size: 14px;
  box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, .1);
}
.Calculator-holder {
  margin-top: 6px;
  padding: 0px 4px;
}
.Calculator-button {
  button {
    width: 100%;
  }
}
.Calculator-order-summary {
  font-weight: bold;
  font-size: 16px;
}

.Calculator-div {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 6px 0px;
}
.Calculator-divider {
  margin: 10px 0px;
  div {
    border-bottom: 1px solid grey;
  }
}
.Calculator-total-summary {
  font-weight: bold;
  color: red;
  font-size: 16px;
}


/**
 * The CSS shown here will not be introduced in the Quickstart guide, but shows
 * how you can use CSS to style your Element's container.
 */
.StripeElement {
  background-color: white;
  padding: 8px 12px;
  border-radius: 4px;
  border: 1px solid transparent;
  box-shadow: 0 1px 3px 0 #e6ebf1;
  -webkit-transition: box-shadow 150ms ease;
  transition: box-shadow 150ms ease;
}

.StripeElement--focus {
  box-shadow: 0 1px 3px 0 #cfd7df;
}

.StripeElement--invalid {
  border-color: #fa755a;
}

.StripeElement--webkit-autofill {
  background-color: #fefde5 !important;
}


@media only screen and (max-width: 700px) {
  .Calculator-container {
    // max-width: 300px;
    // min-width: 300px;
    // height: 320px;
    // display: flex;
    // flex-direction: column;
    // padding: 14px 16px;
    // font-size: 14px;
    // box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, .1);
    margin: 0 auto;
    flex: 1;
    display: flex;
    flex-direction: column;
    padding: 0px 0px 10px;
    min-width: 90%;
    max-width: 90%;
    width: 90%;
  }
  .Calculator-holder {
    margin-top: 6px;
    padding: 0px 24px;
  }
  .Calculator-button {
    button {
      width: 100%;
    }
  }
  .Calculator-order-summary {
    font-weight: bold;
    font-size: 16px;
  }

  .Calculator-div {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin: 6px 0px;
  }
  .Calculator-divider {
    margin: 10px 0px;
    div {
      border-bottom: 1px solid grey;
    }
  }
  .Calculator-total-summary {
    font-weight: bold;
    color: red;
    font-size: 16px;
  }
}
