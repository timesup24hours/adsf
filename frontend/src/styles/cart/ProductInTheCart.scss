.ProductInTheCart-container {
  display: flex;
  flex-direction: row;
  font-size: 14px;
  padding: 10px 20px 10px 0px;
  justify-content: space-between;
  align-items: center;

}

.ProductInTheCart-info-container {
  display: flex;
  flex-direction: row;
  min-width: 300px;
  padding-right: 10px;
  position: relative;
  width: 350px;
  margin-right: 10px;

  .ProductInTheCart-img {
    padding: 10px;
    img {
      max-height: 100px;
      max-width: 80px;
      padding: 10px;
    }
  }

  &:last-child {
    padding-bottom: 40px;
  }

}

.ProductInTheCart-info-holder {
  display: flex;
  flex-direction: column;
  .ProductInTheCart-div {
    margin: 2px 0px;
  }

  .ProductInTheCart-info-name {
    font-size: 16px;
    font-weight: bold;
    @extend .wrapword;
  }
  .ProductInTheCart-info-price {
    color: red;
  }
  .ProductInTheCart-info-desc {

  }
  .ProductInTheCart-info-quantity-section {
    .ProductInTheCart-info-quantity {
      float: left;
    }
    .ProductInTheCart-info-quantity-change {
      float: left;
      margin-left: 8px;
      opacity: .5;
      // color: blue;
      // text-decoration: underline;
      &:hover {
        opacity: 1;
        cursor: pointer;
      }
    }
  }

}


.ProductInTheCart-right-session {
  display: flex;
  justify-content: flex-end;
}

.ProductInTheCart-product-deleted {
  width: 100%;
  height: 100%;
  z-index: 1000;
  position: absolute;
  user-select: none;
  color: red;
  background-color: rgba(0,0,0,.1);
  &:hover {
    cursor: default;
  }
  &:hover div {
    display: none;
  }
  div {
    background-color: white;
    padding: 2px 8px;
    border-radius: 2px;
  }

}

@media only screen and (max-width: 700px) {
  .ProductInTheCart-container {
    display: flex;
    flex-direction: column;
    font-size: 14px;
    padding: 0px;
    justify-content: space-between;
    align-items: center;
    margin: 0 auto;
    width: 90%;

    border-bottom: 1px solid rgba(0,0,0,.1);
    margin-bottom: 30px;

    &:last-child {
      margin-bottom: 0px;
      border-bottom: 0px solid rgba(0,0,0,.1);
    }
  }


  .ProductInTheCart-info-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    min-width: 300px;
    padding-right: 0px;
    width: 300px;
    margin-right: 0px;

    .ProductInTheCart-img {
      padding: 0px;
      img {

        max-height: 300px;
        max-width: 300px;
        padding: 0px;
      }
    }

    &:last-child {
      padding-bottom: 40px;
    }

  }

  .ProductInTheCart-info-holder {

    justify-content: center;
    align-items: center;
    padding: 20px;

    .ProductInTheCart-info-name {
      font-size: 16px;
      font-weight: bold;
      text-align: center;
    }

  }

  .ProductInTheCart-right-session {
    margin: 10px auto 20px;
  }
}
