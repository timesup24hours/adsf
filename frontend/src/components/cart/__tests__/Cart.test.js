// npm packages
import configureMockStore from 'redux-mock-store'

// our packages
import CartWrapper, { Cart } from '../Cart'
import Calculator from '../../calculator/Calculator'
import ProductInTheCart from '../ProductInTheCart'

// create mock store
const mockStore = configureMockStore();

const totalQuantity = 1
const cart = {
  cart: [],
  totalQuantity: 0,
  total: 0,
  pending: false,
  success: false,
  fail: false,
  errors: {
  },
  error: false,
  currentProduct: {},
  quantityOfCurrentProduct: 0,
}


const getCartRequest = () => {
  cart.cart.push({
    product: { _id: '1' },
    quantity: 5,
    total: 155,
  });
  cart.cart.push({
    product: { _id: '2' },
    quantity: 5,
    total: 155,
  });
};

const auth = {
  isAuthenticated: true
}

const router = {
  push: jest.fn()
}

const store = mockStore({ cart: cart });

test('# CartWrapper', () => {
  const wrapper = shallow(<CartWrapper totalQuantity={totalQuantity} store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Cart, should render <div>no item</div>', () => {
  const wrapper = shallow(<Cart cart={cart} totalQuantity={0} store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Cart should render nested component # Calculator', () => {
  const component = (
    <Cart
      cart={cart}
      totalQuantity={totalQuantity}
      getCartRequest={getCartRequest}
      auth={auth}
    />
  );
  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // test if has Calculator
  const app = mount(component);
  expect(app.contains(<Calculator cart={cart} />)).toBe(true);

})

test('# CartWrapper calls componentDidMount', () => {
    const component = (
      <CartWrapper
        cart={cart}
        totalQuantity={totalQuantity}
        getCartRequest={getCartRequest}
        auth={auth}
        store={store}
      />
    );
    sinon.spy(CartWrapper.prototype, 'componentDidMount');
    const wrapper = mount(component);
    expect(CartWrapper.prototype.componentDidMount.calledOnce).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });
