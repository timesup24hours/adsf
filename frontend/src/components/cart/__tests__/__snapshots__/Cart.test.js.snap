exports[`test # Cart should render nested component # Calculator 1`] = `
<div
  className="Cart-container">
  <div
    className="Cart-wrapper">
    <div
      className="Cart-order-holder">
      <div
        className="Cart-order-title">
        Review Your Order:
      </div>
    </div>
    <div
      className="Cart-calculator">
      <Calculator
        cart={
          Object {
            "cart": Array [],
            "currentProduct": Object {},
            "error": false,
            "errors": Object {},
            "fail": false,
            "pending": false,
            "quantityOfCurrentProduct": 0,
            "success": false,
            "total": 0,
            "totalQuantity": 0,
          }
        } />
    </div>
  </div>
</div>
`;

exports[`test # Cart, should render <div>no item</div> 1`] = `
<div
  className="Cart-container">
  <div
    className="Cart-no-items">
    No items
  </div>
</div>
`;

exports[`test # CartWrapper 1`] = `
<Cart
  cart={
    Object {
      "cart": Array [],
      "currentProduct": Object {},
      "error": false,
      "errors": Object {},
      "fail": false,
      "pending": false,
      "quantityOfCurrentProduct": 0,
      "success": false,
      "total": 0,
      "totalQuantity": 0,
    }
  }
  getCartRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  totalQuantity={0} />
`;

exports[`test # CartWrapper calls componentDidMount 1`] = `
<Connect(Cart)
  auth={
    Object {
      "isAuthenticated": true,
    }
  }
  cart={
    Object {
      "cart": Array [
        Object {
          "product": Object {
            "_id": "1",
          },
          "quantity": 5,
          "total": 155,
        },
        Object {
          "product": Object {
            "_id": "2",
          },
          "quantity": 5,
          "total": 155,
        },
      ],
      "currentProduct": Object {},
      "error": false,
      "errors": Object {},
      "fail": false,
      "pending": false,
      "quantityOfCurrentProduct": 0,
      "success": false,
      "total": 0,
      "totalQuantity": 0,
    }
  }
  getCartRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  totalQuantity={1}>
  <Cart
    cart={
      Object {
        "cart": Array [
          Object {
            "product": Object {
              "_id": "1",
            },
            "quantity": 5,
            "total": 155,
          },
          Object {
            "product": Object {
              "_id": "2",
            },
            "quantity": 5,
            "total": 155,
          },
        ],
        "currentProduct": Object {},
        "error": false,
        "errors": Object {},
        "fail": false,
        "pending": false,
        "quantityOfCurrentProduct": 0,
        "success": false,
        "total": 0,
        "totalQuantity": 0,
      }
    }
    getCartRequest={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    totalQuantity={0}>
    <div
      className="Cart-container">
      <div
        className="Cart-no-items">
        No items
      </div>
    </div>
  </Cart>
</Connect(Cart)>
`;
